lib/collection/src/collection_manager/tests/mod.rs

```diff
@@ -92,11 +104,35 @@ fn test_move_points_to_copy_on_write() {
 
     let proxy_id = wrap_proxy(segments.clone(), sid1, dir.path());
 
-    let vectors = vec![vec![0.0, 0.0, 0.0, 0.0], vec![0.0, 0.0, 0.0, 0.0]];
-    upsert_points(&segments, 1001, &[1.into(), 2.into()], &vectors, &None).unwrap();
-
-    let vectors = vec![vec![0.0, 0.0, 0.0, 0.0], vec![0.0, 0.0, 0.0, 0.0]];
-    upsert_points(&segments, 1002, &[2.into(), 3.into()], &vectors, &None).unwrap();
+    let points = vec![
+        PointStruct {
+            id: 1.into(),
+            vector: vec![0.0, 0.0, 0.0, 0.0],
+            payload: None,
+        },
+        PointStruct {
+            id: 2.into(),
+            vector: vec![0.0, 0.0, 0.0, 0.0],
+            payload: None,
+        },
+    ];
+
+    upsert_points(&segments.read(), 1001, &points).unwrap();
+
+    let points = vec![
+        PointStruct {
+            id: 2.into(),
+            vector: vec![0.0, 0.0, 0.0, 0.0],
+            payload: None,
+        },
+        PointStruct {
+            id: 3.into(),
+            vector: vec![0.0, 0.0, 0.0, 0.0],
+            payload: None,
+        },
+    ];
+
+    upsert_points(&segments.read(), 1002, &points).unwrap();
```

Comment:
Maybe also add a test for `sync_points` here?

Reply:
there are no any special new methods for proxy segment, except for `read_range`, which is already tested for proxy